src/optimization/telemetry_analyzer.py

```
@@ -0,0 +1,240 @@
+                    "model": row["model"],
+                    "total_calls": row["total_calls"],
+                    "success_rate": round(success_rate, 3),
+                    "avg_latency_ms": round(row["avg_latency_ms"] or 5000, 1),
+                    "avg_tps": round(row["avg_tps"] or 0, 1),
+                }
+
+            conn.close()
+            logger.info(f"Loaded telemetry from SQLite: {len(providers)} entries")
+            return {"providers": providers, "generated_at": datetime.now().isoformat()}
+
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)